src/test/clojure/xtdb/compactor_test.clj

```diff
@@ -242,13 +256,16 @@
 
           "L2H + L1H take us over the limit, compact early")))
 
+(defn- filter-result [required-level part-prefix res]
+  (filter (comp (fn [{:keys [level part]}]
+                  (and (= level required-level) (= part-prefix (take (count part-prefix) (vec part)))))
+                trie/parse-trie-key
+                :out-trie-key)
+          res))
+
 (t/deftest test-l2+-compaction-jobs
   (binding [cat/*file-size-target* 16]
-    (t/is (= #{ ;; L2 [0] is full, compact L3 [0 0] and [0 1]
-               (job "l03-rc-p00-b0f" ["l02-rc-p0-b03" "l02-rc-p0-b07" "l02-rc-p0-b0b" "l02-rc-p0-b0f"] [0 0])
-               (job "l03-rc-p01-b0f" ["l02-rc-p0-b03" "l02-rc-p0-b07" "l02-rc-p0-b0b" "l02-rc-p0-b0f"] [0 1])
```

Comment:
Added another case for this scenario as `l03` will only happen if the part groups are full.